js/gdm/authPrompt.js

```diff
@@ -600,18 +613,33 @@ const AuthPrompt = new Lang.Class({
         this.emit('cancelled');
     },
 
+    _loadCustomerSupportFile: function() {
```

Comment:
Optional comment, but if this function was called e.g. `_ensureCustomerSupportKeyFile()`, and simply tried to create `this._customerSupportKeyFile` when it did not exist without returning anything, you could swap out the `if (keyFile)` checks in the callers for `if (this._customerSupportKeyFile)` checks, with a net reduction of lines of code :)

Reply:
Considering the search path involves presumably read-only locations (`PKGDATADIR` and `LOCALSTATEDIR` which I assume is reset on reboot?) does it make sense to try and create this file?

Reply:
@senko With "create" I mean "create the instance", not the file on disk.
Basically since the callers of `_loadCustomerSupportFile()` already need to check its return value, and that function already checks the value of the internal `this._customerSupportKeyFile` state variable, I'd remove the latter in favor of checking the internal field in the callers.

Reply:
@senko Ping? I think this is the last pending review comment left to be addressed.

Also, I'm tagging @ramcq here in case he want to comment too, as he explicitly mentioned last week he'd like to take a look to the way the salt is being implemented with this PR.